common/traits/000_wc_leadership_traits.txt

```diff
@@ -0,0 +1,218 @@
+		days_of_supply = 10
+		actual_light_infantry = 0.2
+		archmages = 0.2
+	}
+}
+
+marksman_captain_leader = {
+    leader = yes
+
+    command_modifier = {
+		archers = 0.30
+    }
+}
+
+aerial_raider_leader = {
+    leader = yes
+
+    command_modifier = {
+        knights = 0.20
+		pursue = 0.10
+    }
+}
+
+artillery_engineer_leader = {
+    leader = yes
+
+    command_modifier = {
+        siege_machines = 0.20
+        war_machines = 0.20
+    }
+
+	siege_speed = 0.15
+}
+
+taskmaster_leader = {
+    leader = yes
+	random = no
+
+    command_modifier = {
+        thralls = 0.20
+		speed = 0.10
+		supply_limit = 3
+    }
+}
+
+warpriest_leader = {
+    leader = yes
+	random = no
+	customizer = no
+
+	attrition = -0.10
+
+    command_modifier = {
+        morale_defence = 0.10
+		religious_enemy = 0.10
+        # max_attrition = -0.50
+        # days_of_supply = 30
+    }
+}
+
+offensive_caster_leader = {
+    leader = yes
+	random = no
+	customizer = no
+
+    command_modifier = {
+		actual_light_infantry = 0.1
+        damage = 0.10
+		morale_offence = 0.10
+    }
+}
+
+defensive_caster_leader = {
+    leader = yes
+	random = no
+	customizer = no
+
+	command_modifier = {
+		actual_light_infantry = 0.1
+        defence = 0.10
+		retreat = 0.25
+    }
+}
+
+defiler_leader = {
+    leader = yes
+	trait_effect_capture_commanders = 1
+	random = no
+	customizer = no
+
+    command_modifier = {
+        morale_offence = 0.30
+		pursue = 0.20
+    }
+
+	is_visible = {
+		FROM = {
+			ROOT = {
+				OR = {
+					prev_is_visible_warlock_to_this_trigger = yes
+					prev_is_visible_necromancer_to_this_trigger = yes
+					prev_is_visible_shadow_priest_to_this_trigger = yes
+				}
+			}
+		}
+	}
+}
+
+### Vanilla ones ###
+
+light_foot_leader = {
+	leader = yes
+
+	command_modifier = {
+		actual_heavy_infantry = 0.20
+		morale_defence = 0.10	#Enheartened backbone foot soldiers, enheartened army
+	}
+}
+
+cavalry_leader = {
+	leader = yes
+
+	command_modifier = {
+		morale_offence = 0.1	#Improved cavalry charges, heavy spike to morale
+		light_cavalry = 0.2
+	}
+}
+
+inspiring_leader = {
+	leader = yes
+
+	command_modifier = {
+		morale_offence = 0.15
+		morale_defence = 0.15
+	}
+}
+
+trickster = {
+	leader = yes
+
+	command_modifier = {
+		random = 0.3
+	}
+}
+
+organizer = {
+	leader = yes
+
+	command_modifier = {
+		speed = 0.2
+		retreat = 0.1
+	}
+}
+
+defensive_leader = {
+	leader = yes
+
+	command_modifier = {
+		defence = 0.25
+		damage = -0.1
+	}
+}
+
+# Renamed to "Direct Leader"
+experimenter = {
+	leader = yes
+
+	command_modifier = {
+		center = 0.2
+	}
+}
+
+flanker = {
+	leader = yes
+
+	command_modifier = {
+		flank = 0.2
+	}
+}
+
+aggressive_leader = {
+	leader = yes
+
+	command_modifier = {
+		pursue = 0.25
+		damage = 0.1
+		defence = -0.1
+	}
+}
+
+siege_leader = {
+	leader = yes
+
+	command_modifier = {
+		siege = 0.4
+	}
+}
+
+heavy_infantry_leader = {
+    leader = yes
+
+    command_modifier = {
+        pikemen = 0.20
```

Comment:
Now it can be changed to heavy_infantry, trait loc can be changed to "Melee Specialist" or something like that